sugarscape.py

```diff
@@ -674,15 +674,17 @@ def toggleRun(self):
         self.run = not self.run
 
     def updateGiniCoefficient(self):
-        agentWealths = sorted([agent.wealth for agent in self.agents])
-        # Calculate area between line of equality and Lorenz curve of agent wealths
-        height = 0
-        area = 0
+        agentWealths = sorted([agent.sugar + agent.spice for agent in self.agents])
+        # Calculate normalized area of Lorenz curve of agent wealths
+        totalWealth = sum(agentWealths)
+        cumulativeWealth = 0
+        lorenzCurveArea = 0
         for wealth in agentWealths:
-            height += wealth
-            area += (height - wealth) / 2
-        lineOfEquality = (height * len(agentWealths)) / 2
-        giniCoefficient = round((lineOfEquality - area) / max(1, lineOfEquality), 2)
+            cumulativeWealth += wealth
+            lorenzCurveArea += cumulativeWealth / totalWealth
+        lorenzCurveArea /= len(agentWealths)
+
+        giniCoefficient = round((0.5 - lorenzCurveArea) / 0.5, 2)
```

Comment:
We can keep the magic number since the two areas we care about add up to 0.5. Put that in a variable called `areaUnderLineOfEquality` or similarly descriptive variable and add a _brief_ comment explaining the value.

Looking at the book, let's also round to 3 rather than 2 decimal places.